app/src/main/java/pub/devrel/easypermissions/sample/MainActivity.java

```diff
@@ -58,6 +59,18 @@ public void onClick(View v) {
         });
     }
 
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == EasyPermissions.SETTINGS_REQ_CODE) {
```

Comment:
Actually that's not a dialog listener :)
The listener i added is for the negative button only. If positive button clicked, user won't get notified, they will be redirected to the settings screen (the library handle it, as the previous behaviour). So, this `onActivityResult` is for listening whether the user is from settings screen or not. User needs to check the permissions that may be changed during in the settings screen manually. I'll update the sample code to make it clearer.